x/auth/tx/service_test.go

```diff
@@ -139,14 +156,59 @@ func (s IntegrationTestSuite) TestGetTxEvents() {
 	s.Require().Equal(len(grpcRes.Txs), 1)
 	s.Require().Equal("foobar", grpcRes.Txs[0].Body.Memo)
 
-	// Query the tx via grpc-gateway.
-	restRes, err := rest.GetRequest(fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?event=%s&pagination.offset=%d&pagination.limit=%d", val.APIAddress, "message.action=send", 0, 1))
-	s.Require().NoError(err)
-	var getTxRes tx.GetTxsEventResponse
-	s.Require().NoError(val.ClientCtx.JSONMarshaler.UnmarshalJSON(restRes, &getTxRes))
-	s.Require().Equal(len(grpcRes.Txs), 1)
-	s.Require().Equal("foobar", getTxRes.Txs[0].Body.Memo)
-	s.Require().NotZero(grpcRes.TxResponses[0].Height)
+	rpcTests := []struct {
+		name      string
+		url       string
+		expectErr bool
+		expected  *tx.GetTxsEventResponse
+	}{
+		{
+			"empty params",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs", val.APIAddress),
+			true,
+			&tx.GetTxsEventResponse{},
+		},
+		{
+			"without pagination",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?events=%s", val.APIAddress, "message.action=send"),
+			false,
+			&tx.GetTxsEventResponse{},
+		},
+		{
+			"with pagination",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?events=%s&pagination.offset=%d&pagination.limit=%d", val.APIAddress, "message.action=send", 0, 10),
+			false,
+			&tx.GetTxsEventResponse{},
+		},
+		{
+			"expect pass with multiple-events",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?events=%s&events=%s", val.APIAddress, "message.action=send", "message.module=bank"),
+			false,
+			&tx.GetTxsEventResponse{},
+		},
+		{
+			"expect pass with escape event",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?events=%s", val.APIAddress, "message.action%3Dsend"),
+			false,
+			&tx.GetTxsEventResponse{},
```

Comment:
we could remove this line from all test cases, and just use a local `var res tx.GetTxsEventResponse` and unmarshal into `&res`.